product-mini/platforms/posix/main.c

```diff
@@ -8,6 +8,9 @@
 #endif
 #include <stdlib.h>
 #include <string.h>
+#if defined(_POSIX_SOURCE) || defined(_POSIX_C_SOURCE)
```

Comment:
what's the intention of this `defined(_POSIX_SOURCE) || defined(_POSIX_C_SOURCE)`?
is it meant to be an availability check of dlopen?

on nuttx, the availability of dlopen depends on a build-time config.
i feel it's more straightforward to control this with BH_HAS_DLFCN or something like that.
how do you think?

Reply:
https://github.com/bytecodealliance/wasm-micro-runtime/pull/1134